wayland-logout.c

```diff
@@ -35,64 +35,84 @@
 #include <sys/socket.h>
 #include <sys/un.h>
 
+// size of field sun_path in struct stsockaddr_un
+#define pathlen ((int)sizeof(((struct sockaddr_un *)NULL)->sun_path))
+
 int main()
 {
-    char socket_path[108];
+    char socket_path[pathlen];
     char *xdg_runtime_dir = getenv("XDG_RUNTIME_DIR");
     char *wayland_display = getenv("WAYLAND_DISPLAY");
 
-    if (!xdg_runtime_dir)
+    if (!xdg_runtime_dir && !wayland_display)
     {
-        printf("XDG_RUNTIME_DIR not set\n");
+        fprintf(stderr, "XDG_RUNTIME_DIR and WAYLAND_DISPLAY not set\n");
         return -1;
     }
-    if (!wayland_display)
+    else if (wayland_display)
     {
-        printf("WAYLAND_DISPLAY not set\n");
-        return -1;
+        if (wayland_display[0] == '/')
+        {
+            // WAYLAND_DISPLAY can be an absolute path
+            if (snprintf(socket_path, pathlen, "%s", wayland_display) >= pathlen)
```

Comment:
```suggestion
            if (snprintf(socket_path, pathlen, "%s", wayland_display) == pathlen)
```

"The functions snprintf() and vsnprintf() write at most size bytes (including the terminating null byte ('\0')) to str." per printf(3).

Reply:
The important part is right after that (emphasis mine):

> If the output was
       truncated due to this limit, then the return value is the number of
       characters (excluding the terminating null byte) which would have been
       written to the final string if enough space had been available.  *Thus,
       a return value of size or more means that the output was truncated.*
       (See also below under NOTES.)

Reply:
I can leave a comment as a reminder there.

Reply:
No worries, just making sure you're RTFM.